./configure fails at with "Permission denied" error

I just downloaded `open-cas-linux` project and follow instructions to compile it.

Unfortunately `./configure` produces a lot of error messages and then fails with a cryptic one:

```
... about a thousand of following messaes...
mkdir: cannot create directory ‘.tmp_1056416’: Permission denied
mkdir: cannot create directory ‘.tmp_1056420’: Permission denied
mkdir: cannot create directory ‘.tmp_1056421’: Permission denied
mkdir: cannot create directory ‘.tmp_1056423’: Permission denied
mkdir: cannot create directory ‘.tmp_1056425’: Permission denied
mkdir: cannot create directory ‘.tmp_1056427’: Permission denied
mkdir: cannot create directory ‘.tmp_1056430’: Permission denied
mkdir: cannot create directory ‘.tmp_1056433’: Permission denied
mkdir: cannot create directory ‘.tmp_1056435’: Permission denied
mkdir: cannot create directory ‘.tmp_1056439’: Permission denied
mkdir: cannot create directory ‘.tmp_1056441’: Permission denied
mkdir: cannot create directory ‘.tmp_1056445’: Permission denied
mkdir: cannot create directory ‘.tmp_1056447’: Permission denied
mkdir: cannot create directory ‘.tmp_1056452’: Permission denied
mkdir: cannot create directory ‘.tmp_1056453’: Permission denied
mkdir: cannot create directory ‘.tmp_1056458’: Permission denied
mkdir: cannot create directory ‘.tmp_1056457’: Permission denied
mkdir: cannot create directory ‘.tmp_1056463’: Permission denied
ERROR! Following steps failed while preparing config:
1_generic_acct.conf
```

I enabled more bash logging with `bash -x ./configure` and here are additional logs from around the "Permission denied" error:

```
++ IFS='?'
++ case "$1" in
++ check /mnt/cold/sources/open-cas-linux/config.out /mnt/cold/sources/open-cas-linux/configure.d/2_queue_write.conf
+++ basename /mnt/cold/sources/open-cas-linux/configure.d/2_queue_write.conf
++ cur_name=2_queue_write.conf
++ config_file_path=/mnt/cold/sources/open-cas-linux/config.out
++ compile_module 2_queue_write.conf 'blk_queue_write_cache(NULL, 0, 0)' linux/blkdev.h
++ '[' 3 -gt 2 ']'
++ i=3
++ '[' 3 -le 3 ']'
+++ echo -e '\n#include <linux/blkdev.h>\n'
++ INCLUDE+='
#include <linux/blkdev.h>'
++ i=4
++ '[' 4 -le 3 ']'
++ config_file=2_queue_write.conf
++ test_module_dir=/mnt/cold/sources/open-cas-linux/configure.d/2_queue_write.conf_dir
++ test_module_log=/mnt/cold/sources/open-cas-linux/configure.d/.2_queue_write.conf.log
++ test_module_file=/mnt/cold/sources/open-cas-linux/configure.d/2_queue_write.conf_dir/test_mod.c
++ test_module_obj=/mnt/cold/sources/open-cas-linux/configure.d/2_queue_write.conf_dir/test_mod.o
++ mkdir -p /mnt/cold/sources/open-cas-linux/configure.d/2_queue_write.conf_dir
++ cp -f /mnt/cold/sources/open-cas-linux/configure.d/Makefile /mnt/cold/sources/open-cas-linux/configure.d/2_queue_write.conf_dir
++ cat
++ echo '### blk_queue_write_cache(NULL, 0, 0) ###'
++ make -C /mnt/cold/sources/open-cas-linux/configure.d/2_queue_write.conf_dir KERNEL_DIR=/lib/modules/5.8.12-arch1-1/build/
++ local ret=0
++ rm -Rf /mnt/cold/sources/open-cas-linux/configure.d/1_wtlh.conf_dir
++ '[' 0 -eq 0 ']'
++ rm -f /mnt/cold/sources/open-cas-linux/configure.d/.1_wtlh.conf.log
++ return 0
++ echo 1_wtlh.conf 1
++ IFS=' 	
'
++ local ret=0
++ rm -Rf /mnt/cold/sources/open-cas-linux/configure.d/2_bio_barrier.conf_dir
++ '[' 0 -eq 0 ']'
++ rm -f /mnt/cold/sources/open-cas-linux/configure.d/.2_bio_barrier.conf.log
++ return 0
++ echo 2_bio_barrier.conf 1
++ IFS=' 	
'
++ local ret=0
++ rm -Rf /mnt/cold/sources/open-cas-linux/configure.d/1_write_fua.conf_dir
++ '[' 0 -eq 0 ']'
++ rm -f /mnt/cold/sources/open-cas-linux/configure.d/.1_write_fua.conf.log
++ return 0
++ echo 1_write_fua.conf 3
++ IFS=' 	
'
++ ps --no-headers -o pid --ppid=1065347
++ wc -w
+ '[' 5 -ge 8 ']'
+ for file in $CONFIG_FILES
+ source /mnt/cold/sources/open-cas-linux/configure.d/2_write_flush.conf check /mnt/cold/sources/open-cas-linux/config.out /mnt/cold/sources/open-cas-linux/configure.d/2_write_flush.conf
+++ dirname /mnt/cold/sources/open-cas-linux/configure.d/2_write_flush.conf
++ ps --no-headers -o pid --ppid=1065347
++ . /mnt/cold/sources/open-cas-linux/configure.d/conf_framework
++ wc -w
++++ dirname ./configure
+++ SCRIPTPATH=.
++++ realpath .
+++ SCRIPTPATH=/mnt/cold/sources/open-cas-linux
++++ uname -r
+++ KERNEL_DIR=/lib/modules/5.8.12-arch1-1/build/
++++ cd /lib/modules/5.8.12-arch1-1/build/
++++ make kernelversion
+ '[' 6 -ge 8 ']'
+ wait
mkdir: cannot create directory ‘.tmp_1089863’: Permission denied
mkdir: cannot create directory ‘.tmp_1089867’: Permission denied
mkdir: cannot create directory ‘.tmp_1089869’: Permission denied
mkdir: cannot create directory ‘.tmp_1089871’: Permission denied
mkdir: cannot create directory ‘.tmp_1089873’: Permission denied
mkdir: cannot create directory ‘.tmp_1089875’: Permission denied
mkdir: cannot create directory ‘.tmp_1089878’: Permission denied
mkdir: cannot create directory ‘.tmp_1089880’: Permission denied
mkdir: cannot create directory ‘.tmp_1089882’: Permission denied
mkdir: cannot create directory ‘.tmp_1089884’: Permission denied
mkdir: cannot create directory ‘.tmp_1089886’: Permission denied
mkdir: cannot create directory ‘.tmp_1089888’: Permission denied
mkdir: cannot create directory ‘.tmp_1089890’: Permission denied
mkdir: cannot create directory ‘.tmp_1089892’: Permission denied
mkdir: cannot create directory ‘.tmp_1089894’: Permission denied
mkdir: cannot create directory ‘.tmp_1089896’: Permission denied
mkdir: cannot create directory ‘.tmp_1089898’: Permission denied
mkdir: cannot create directory ‘.tmp_1089900’: Permission denied
mkdir: cannot create directory ‘.tmp_1089902’: Permission denied
mkdir: cannot create directory ‘.tmp_1089904’: Permission denied
mkdir: cannot create directory ‘.tmp_1089906’: Permission denied
mkdir: cannot create directory ‘.tmp_1089908’: Permission denied
mkdir: cannot create directory ‘.tmp_1089910’: Permission denied
mkdir: cannot create directory ‘.tmp_1089912’: Permission denied
mkdir: cannot create directory ‘.tmp_1089914’: Permission denied
mkdir: cannot create directory ‘.tmp_1089916’: Permission denied
mkdir: cannot create directory ‘.tmp_1089918’: Permission denied
mkdir: cannot create directory ‘.tmp_1089920’: Permission denied
mkdir: cannot create directory ‘.tmp_1089922’: Permission denied
mkdir: cannot create directory ‘.tmp_1089924’: Permission denied
mkdir: cannot create directory ‘.tmp_1089927’: Permission denied
mkdir: cannot create directory ‘.tmp_1089929’: Permission denied
mkdir: cannot create directory ‘.tmp_1089931’: Permission denied
mkdir: cannot create directory ‘.tmp_1089933’: Permission denied
mkdir: cannot create directory ‘.tmp_1089935’: Permission denied
mkdir: cannot create directory ‘.tmp_1089937’: Permission denied
mkdir: cannot create directory ‘.tmp_1089939’: Permission denied
mkdir: cannot create directory ‘.tmp_1089941’: Permission denied
mkdir: cannot create directory ‘.tmp_1089943’: Permission denied
mkdir: cannot create directory ‘.tmp_1089945’: Permission denied
mkdir: cannot create directory ‘.tmp_1089947’: Permission denied
mkdir: cannot create directory ‘.tmp_1089949’: Permission denied
mkdir: cannot create directory ‘.tmp_1089951’: Permission denied
mkdir: cannot create directory ‘.tmp_1089953’: Permission denied
mkdir: cannot create directory ‘.tmp_1089955’: Permission denied
mkdir: cannot create directory ‘.tmp_1089957’: Permission denied
+++ KERNEL_VER=5.8.12-arch1
++++ nproc
+++ NPROC=8
+++ DEFINE_FILE=/mnt/cold/sources/open-cas-linux/modules/generated_defines.h
++ conf_run check /mnt/cold/sources/open-cas-linux/config.out /mnt/cold/sources/open-cas-linux/configure.d/2_write_flush.conf
++ local 'OLD_IFS= 	
'
++ IFS='?'
++ case "$1" in
++ check /mnt/cold/sources/open-cas-linux/config.out /mnt/cold/sources/open-cas-linux/configure.d/2_write_flush.conf
+++ basename /mnt/cold/sources/open-cas-linux/configure.d/2_write_flush.conf
++ cur_name=2_write_flush.conf
++ config_file_path=/mnt/cold/sources/open-cas-linux/config.out
++ compile_module 2_write_flush.conf WRITE_FLUSH linux/fs.h

......


++ config_file=2_write_flush.conf
++ test_module_dir=/mnt/cold/sources/open-cas-linux/configure.d/2_write_flush.conf_dir
++ test_module_log=/mnt/cold/sources/open-cas-linux/configure.d/.2_write_flush.conf.log
++ test_module_file=/mnt/cold/sources/open-cas-linux/configure.d/2_write_flush.conf_dir/test_mod.c
++ test_module_obj=/mnt/cold/sources/open-cas-linux/configure.d/2_write_flush.conf_dir/test_mod.o
++ mkdir -p /mnt/cold/sources/open-cas-linux/configure.d/2_write_flush.conf_dir
++ cp -f /mnt/cold/sources/open-cas-linux/configure.d/Makefile /mnt/cold/sources/open-cas-linux/configure.d/2_write_flush.conf_dir
++ cat
++ echo '### REQ_PREFLUSH ###'
++ make -C /mnt/cold/sources/open-cas-linux/configure.d/2_write_flush.conf_dir KERNEL_DIR=/lib/modules/5.8.12-arch1-1/build/
++ local ret=0
++ rm -Rf /mnt/cold/sources/open-cas-linux/configure.d/2_flush_fua.conf_dir
++ '[' 0 -eq 0 ']'
++ rm -f /mnt/cold/sources/open-cas-linux/configure.d/.2_flush_fua.conf.log
++ return 0
++ echo 2_flush_fua.conf 3
++ IFS=' 	
'
++ local ret=0
++ rm -Rf /mnt/cold/sources/open-cas-linux/configure.d/2_write_flush.conf_dir
++ '[' 0 -eq 0 ']'
++ rm -f /mnt/cold/sources/open-cas-linux/configure.d/.2_write_flush.conf.log
++ return 0
++ echo 2_write_flush.conf 3
++ IFS=' 	
'
+ grep X /mnt/cold/sources/open-cas-linux/config.out
+ '[' 0 -eq 0 ']'
+ echo 'ERROR! Following steps failed while preparing config:'
ERROR! Following steps failed while preparing config:
+ grep X /mnt/cold/sources/open-cas-linux/config.out
+ cut -f1 -d ' '
1_generic_acct.conf
+ exit 1
```

@imjfckm with the linux `make kernelversion` workaround the build still fails with

```
+ for file in $CONFIG_FILES
+ source /mnt/cold/sources/open-cas-linux/configure.d/2_flush_fua.conf check /mnt/cold/sources/open-cas-linux/config.out /mnt/cold/sources/open-cas-linux/configure.d/2_flush_fua.conf
+++ dirname /mnt/cold/sources/open-cas-linux/configure.d/2_flush_fua.conf
++ . /mnt/cold/sources/open-cas-linux/configure.d/conf_framework
++++ dirname ./configure
+++ SCRIPTPATH=.
++++ realpath .
+++ SCRIPTPATH=/mnt/cold/sources/open-cas-linux
+++ KERNEL_DIR=/lib/modules/5.8.12-arch1-1/build/
++++ make -C /lib/modules/5.8.12-arch1-1/build/ kernelversion
+++ KERNEL_VER='make: Entering directory '\''/usr/lib/modules/5.8.12-arch1-1/build'\''
5.8.12-arch1
make: Leaving directory '\''/usr/lib/modules/5.8.12-arch1-1/build'\'''
++++ nproc
+++ NPROC=8
+++ DEFINE_FILE=/mnt/cold/sources/open-cas-linux/modules/generated_defines.h
++ conf_run check /mnt/cold/sources/open-cas-linux/config.out /mnt/cold/sources/open-cas-linux/configure.d/2_flush_fua.conf
++ local 'OLD_IFS= 	
'
++ IFS='?'
++ case "$1" in
++ check /mnt/cold/sources/open-cas-linux/config.out /mnt/cold/sources/open-cas-linux/configure.d/2_flush_fua.conf
+++ basename /mnt/cold/sources/open-cas-linux/configure.d/2_flush_fua.conf
++ cur_name=2_flush_fua.conf
++ config_file_path=/mnt/cold/sources/open-cas-linux/config.out
++ compile_module 2_flush_fua.conf WRITE_FLUSH_FUA linux/fs.h
++ '[' 3 -gt 2 ']'
++ i=3
++ '[' 3 -le 3 ']'
+++ echo -e '\n#include <linux/fs.h>\n'
++ INCLUDE+='
#include <linux/fs.h>'
++ i=4
++ '[' 4 -le 3 ']'
++ config_file=2_flush_fua.conf
++ test_module_dir=/mnt/cold/sources/open-cas-linux/configure.d/2_flush_fua.conf_dir
++ test_module_log=/mnt/cold/sources/open-cas-linux/configure.d/.2_flush_fua.conf.log
++ test_module_file=/mnt/cold/sources/open-cas-linux/configure.d/2_flush_fua.conf_dir/test_mod.c
++ test_module_obj=/mnt/cold/sources/open-cas-linux/configure.d/2_flush_fua.conf_dir/test_mod.o
++ mkdir -p /mnt/cold/sources/open-cas-linux/configure.d/2_flush_fua.conf_dir
++ cp -f /mnt/cold/sources/open-cas-linux/configure.d/Makefile /mnt/cold/sources/open-cas-linux/configure.d/2_flush_fua.conf_dir
++ cat
++ echo '### WRITE_FLUSH_FUA ###'
++ make -C /mnt/cold/sources/open-cas-linux/configure.d/2_flush_fua.conf_dir KERNEL_DIR=/lib/modules/5.8.12-arch1-1/build/
++ local ret=2
++ rm -Rf /mnt/cold/sources/open-cas-linux/configure.d/2_flush_fua.conf_dir
++ '[' 2 -eq 0 ']'
++ return 2
++ compile_module 2_flush_fua.conf REQ_PREFLUSH linux/blk_types.h
++ '[' 3 -gt 2 ']'
++ i=3
++ '[' 3 -le 3 ']'
+++ echo -e '\n#include <linux/blk_types.h>\n'
++ INCLUDE+='
#include <linux/blk_types.h>'
++ i=4
++ '[' 4 -le 3 ']'
++ config_file=2_flush_fua.conf
++ test_module_dir=/mnt/cold/sources/open-cas-linux/configure.d/2_flush_fua.conf_dir
++ test_module_log=/mnt/cold/sources/open-cas-linux/configure.d/.2_flush_fua.conf.log
++ test_module_file=/mnt/cold/sources/open-cas-linux/configure.d/2_flush_fua.conf_dir/test_mod.c
++ test_module_obj=/mnt/cold/sources/open-cas-linux/configure.d/2_flush_fua.conf_dir/test_mod.o
++ mkdir -p /mnt/cold/sources/open-cas-linux/configure.d/2_flush_fua.conf_dir
++ cp -f /mnt/cold/sources/open-cas-linux/configure.d/Makefile /mnt/cold/sources/open-cas-linux/configure.d/2_flush_fua.conf_dir
++ cat
++ echo '### REQ_PREFLUSH ###'
++ make -C /mnt/cold/sources/open-cas-linux/configure.d/2_flush_fua.conf_dir KERNEL_DIR=/lib/modules/5.8.12-arch1-1/build/
++ local ret=0
++ rm -Rf /mnt/cold/sources/open-cas-linux/configure.d/2_flush_fua.conf_dir
++ '[' 0 -eq 0 ']'
++ rm -f /mnt/cold/sources/open-cas-linux/configure.d/.2_flush_fua.conf.log
++ return 0
++ echo 2_flush_fua.conf 3
++ IFS=' 	
'
+ for file in $CONFIG_FILES
+ source /mnt/cold/sources/open-cas-linux/configure.d/2_make_req.conf check /mnt/cold/sources/open-cas-linux/config.out /mnt/cold/sources/open-cas-linux/configure.d/2_make_req.conf
+++ dirname /mnt/cold/sources/open-cas-linux/configure.d/2_make_req.conf
++ . /mnt/cold/sources/open-cas-linux/configure.d/conf_framework
++++ dirname ./configure
+++ SCRIPTPATH=.
++++ realpath .
+++ SCRIPTPATH=/mnt/cold/sources/open-cas-linux
+++ KERNEL_DIR=/lib/modules/5.8.12-arch1-1/build/
++++ make -C /lib/modules/5.8.12-arch1-1/build/ kernelversion
+++ KERNEL_VER='make: Entering directory '\''/usr/lib/modules/5.8.12-arch1-1/build'\''
5.8.12-arch1
make: Leaving directory '\''/usr/lib/modules/5.8.12-arch1-1/build'\'''
++++ nproc
+++ NPROC=8
+++ DEFINE_FILE=/mnt/cold/sources/open-cas-linux/modules/generated_defines.h
++ conf_run check /mnt/cold/sources/open-cas-linux/config.out /mnt/cold/sources/open-cas-linux/configure.d/2_make_req.conf
++ local 'OLD_IFS= 	
'
++ IFS='?'
++ case "$1" in
++ check /mnt/cold/sources/open-cas-linux/config.out /mnt/cold/sources/open-cas-linux/configure.d/2_make_req.conf
+++ basename /mnt/cold/sources/open-cas-linux/configure.d/2_make_req.conf
++ cur_name=2_make_req.conf
++ config_file_path=/mnt/cold/sources/open-cas-linux/config.out
++ compile_module 2_make_req.conf 'blk_make_request(NULL, NULL, 0)' linux/blkdev.h
++ '[' 3 -gt 2 ']'
++ i=3
++ '[' 3 -le 3 ']'
+++ echo -e '\n#include <linux/blkdev.h>\n'
++ INCLUDE+='
#include <linux/blkdev.h>'
++ i=4
++ '[' 4 -le 3 ']'
++ config_file=2_make_req.conf
++ test_module_dir=/mnt/cold/sources/open-cas-linux/configure.d/2_make_req.conf_dir
++ test_module_log=/mnt/cold/sources/open-cas-linux/configure.d/.2_make_req.conf.log
++ test_module_file=/mnt/cold/sources/open-cas-linux/configure.d/2_make_req.conf_dir/test_mod.c
++ test_module_obj=/mnt/cold/sources/open-cas-linux/configure.d/2_make_req.conf_dir/test_mod.o
++ mkdir -p /mnt/cold/sources/open-cas-linux/configure.d/2_make_req.conf_dir
++ cp -f /mnt/cold/sources/open-cas-linux/configure.d/Makefile /mnt/cold/sources/open-cas-linux/configure.d/2_make_req.conf_dir
++ cat
++ echo '### blk_make_request(NULL, NULL, 0) ###'
++ make -C /mnt/cold/sources/open-cas-linux/configure.d/2_make_req.conf_dir KERNEL_DIR=/lib/modules/5.8.12-arch1-1/build/
++ local ret=2
++ rm -Rf /mnt/cold/sources/open-cas-linux/configure.d/2_make_req.conf_dir
++ '[' 2 -eq 0 ']'
++ return 2
++ echo 2_make_req.conf 2
++ IFS=' 	
'
+ for file in $CONFIG_FILES
+ source /mnt/cold/sources/open-cas-linux/configure.d/2_queue_write.conf check /mnt/cold/sources/open-cas-linux/config.out /mnt/cold/sources/open-cas-linux/configure.d/2_queue_write.conf
+++ dirname /mnt/cold/sources/open-cas-linux/configure.d/2_queue_write.conf
++ . /mnt/cold/sources/open-cas-linux/configure.d/conf_framework
++++ dirname ./configure
+++ SCRIPTPATH=.
++++ realpath .
+++ SCRIPTPATH=/mnt/cold/sources/open-cas-linux
+++ KERNEL_DIR=/lib/modules/5.8.12-arch1-1/build/
++++ make -C /lib/modules/5.8.12-arch1-1/build/ kernelversion
+++ KERNEL_VER='make: Entering directory '\''/usr/lib/modules/5.8.12-arch1-1/build'\''
5.8.12-arch1
make: Leaving directory '\''/usr/lib/modules/5.8.12-arch1-1/build'\'''
++++ nproc
+++ NPROC=8
+++ DEFINE_FILE=/mnt/cold/sources/open-cas-linux/modules/generated_defines.h
++ conf_run check /mnt/cold/sources/open-cas-linux/config.out /mnt/cold/sources/open-cas-linux/configure.d/2_queue_write.conf
++ local 'OLD_IFS= 	
'
++ IFS='?'
++ case "$1" in
++ check /mnt/cold/sources/open-cas-linux/config.out /mnt/cold/sources/open-cas-linux/configure.d/2_queue_write.conf
+++ basename /mnt/cold/sources/open-cas-linux/configure.d/2_queue_write.conf
++ cur_name=2_queue_write.conf
++ config_file_path=/mnt/cold/sources/open-cas-linux/config.out
++ compile_module 2_queue_write.conf 'blk_queue_write_cache(NULL, 0, 0)' linux/blkdev.h
++ '[' 3 -gt 2 ']'
++ i=3
++ '[' 3 -le 3 ']'
+++ echo -e '\n#include <linux/blkdev.h>\n'
++ INCLUDE+='
#include <linux/blkdev.h>'
++ i=4
++ '[' 4 -le 3 ']'
++ config_file=2_queue_write.conf
++ test_module_dir=/mnt/cold/sources/open-cas-linux/configure.d/2_queue_write.conf_dir
++ test_module_log=/mnt/cold/sources/open-cas-linux/configure.d/.2_queue_write.conf.log
++ test_module_file=/mnt/cold/sources/open-cas-linux/configure.d/2_queue_write.conf_dir/test_mod.c
++ test_module_obj=/mnt/cold/sources/open-cas-linux/configure.d/2_queue_write.conf_dir/test_mod.o
++ mkdir -p /mnt/cold/sources/open-cas-linux/configure.d/2_queue_write.conf_dir
++ cp -f /mnt/cold/sources/open-cas-linux/configure.d/Makefile /mnt/cold/sources/open-cas-linux/configure.d/2_queue_write.conf_dir
++ cat
++ echo '### blk_queue_write_cache(NULL, 0, 0) ###'
++ make -C /mnt/cold/sources/open-cas-linux/configure.d/2_queue_write.conf_dir KERNEL_DIR=/lib/modules/5.8.12-arch1-1/build/
++ local ret=0
++ rm -Rf /mnt/cold/sources/open-cas-linux/configure.d/2_queue_write.conf_dir
++ '[' 0 -eq 0 ']'
++ rm -f /mnt/cold/sources/open-cas-linux/configure.d/.2_queue_write.conf.log
++ return 0
++ echo 2_queue_write.conf 1
++ IFS=' 	
'
+ for file in $CONFIG_FILES
+ source /mnt/cold/sources/open-cas-linux/configure.d/2_write_flush.conf check /mnt/cold/sources/open-cas-linux/config.out /mnt/cold/sources/open-cas-linux/configure.d/2_write_flush.conf
+++ dirname /mnt/cold/sources/open-cas-linux/configure.d/2_write_flush.conf
++ . /mnt/cold/sources/open-cas-linux/configure.d/conf_framework
++++ dirname ./configure
+++ SCRIPTPATH=.
++++ realpath .
+++ SCRIPTPATH=/mnt/cold/sources/open-cas-linux
+++ KERNEL_DIR=/lib/modules/5.8.12-arch1-1/build/
++++ make -C /lib/modules/5.8.12-arch1-1/build/ kernelversion
+++ KERNEL_VER='make: Entering directory '\''/usr/lib/modules/5.8.12-arch1-1/build'\''
5.8.12-arch1
make: Leaving directory '\''/usr/lib/modules/5.8.12-arch1-1/build'\'''
++++ nproc
+++ NPROC=8
+++ DEFINE_FILE=/mnt/cold/sources/open-cas-linux/modules/generated_defines.h
++ conf_run check /mnt/cold/sources/open-cas-linux/config.out /mnt/cold/sources/open-cas-linux/configure.d/2_write_flush.conf
++ local 'OLD_IFS= 	
'
++ IFS='?'
++ case "$1" in
++ check /mnt/cold/sources/open-cas-linux/config.out /mnt/cold/sources/open-cas-linux/configure.d/2_write_flush.conf
+++ basename /mnt/cold/sources/open-cas-linux/configure.d/2_write_flush.conf
++ cur_name=2_write_flush.conf
++ config_file_path=/mnt/cold/sources/open-cas-linux/config.out
++ compile_module 2_write_flush.conf WRITE_FLUSH linux/fs.h
++ '[' 3 -gt 2 ']'
++ i=3
++ '[' 3 -le 3 ']'
+++ echo -e '\n#include <linux/fs.h>\n'
++ INCLUDE+='
#include <linux/fs.h>'
++ i=4
++ '[' 4 -le 3 ']'
++ config_file=2_write_flush.conf
++ test_module_dir=/mnt/cold/sources/open-cas-linux/configure.d/2_write_flush.conf_dir
++ test_module_log=/mnt/cold/sources/open-cas-linux/configure.d/.2_write_flush.conf.log
++ test_module_file=/mnt/cold/sources/open-cas-linux/configure.d/2_write_flush.conf_dir/test_mod.c
++ test_module_obj=/mnt/cold/sources/open-cas-linux/configure.d/2_write_flush.conf_dir/test_mod.o
++ mkdir -p /mnt/cold/sources/open-cas-linux/configure.d/2_write_flush.conf_dir
++ cp -f /mnt/cold/sources/open-cas-linux/configure.d/Makefile /mnt/cold/sources/open-cas-linux/configure.d/2_write_flush.conf_dir
++ cat
++ echo '### WRITE_FLUSH ###'
++ make -C /mnt/cold/sources/open-cas-linux/configure.d/2_write_flush.conf_dir KERNEL_DIR=/lib/modules/5.8.12-arch1-1/build/
++ local ret=2
++ rm -Rf /mnt/cold/sources/open-cas-linux/configure.d/2_write_flush.conf_dir
++ '[' 2 -eq 0 ']'
++ return 2
++ compile_module 2_write_flush.conf REQ_PREFLUSH linux/blk_types.h
++ '[' 3 -gt 2 ']'
++ i=3
++ '[' 3 -le 3 ']'
+++ echo -e '\n#include <linux/blk_types.h>\n'
++ INCLUDE+='
#include <linux/blk_types.h>'
++ i=4
++ '[' 4 -le 3 ']'
++ config_file=2_write_flush.conf
++ test_module_dir=/mnt/cold/sources/open-cas-linux/configure.d/2_write_flush.conf_dir
++ test_module_log=/mnt/cold/sources/open-cas-linux/configure.d/.2_write_flush.conf.log
++ test_module_file=/mnt/cold/sources/open-cas-linux/configure.d/2_write_flush.conf_dir/test_mod.c
++ test_module_obj=/mnt/cold/sources/open-cas-linux/configure.d/2_write_flush.conf_dir/test_mod.o
++ mkdir -p /mnt/cold/sources/open-cas-linux/configure.d/2_write_flush.conf_dir
++ cp -f /mnt/cold/sources/open-cas-linux/configure.d/Makefile /mnt/cold/sources/open-cas-linux/configure.d/2_write_flush.conf_dir
++ cat
++ echo '### REQ_PREFLUSH ###'
++ make -C /mnt/cold/sources/open-cas-linux/configure.d/2_write_flush.conf_dir KERNEL_DIR=/lib/modules/5.8.12-arch1-1/build/
++ local ret=0
++ rm -Rf /mnt/cold/sources/open-cas-linux/configure.d/2_write_flush.conf_dir
++ '[' 0 -eq 0 ']'
++ rm -f /mnt/cold/sources/open-cas-linux/configure.d/.2_write_flush.conf.log
++ return 0
++ echo 2_write_flush.conf 3
++ IFS=' 	
'
+ wait
+ grep X /mnt/cold/sources/open-cas-linux/config.out
+ '[' 0 -eq 0 ']'
+ echo 'ERROR! Following steps failed while preparing config:'
ERROR! Following steps failed while preparing config:
+ grep X /mnt/cold/sources/open-cas-linux/config.out
+ cut -f1 -d ' '
1_generic_acct.conf
+ exit 1
```

Looking at ./configure.d/.1_generic_acct.conf.log I see following compilation error:

```
### generic_start_io_acct(NULL, 0, 0, NULL) ###
make: Entering directory '/mnt/cold/sources/open-cas-linux/configure.d/1_generic_acct.conf_dir'
make -C /lib/modules/5.8.12-arch1-1/build/ M=/mnt/cold/sources/open-cas-linux/configure.d/1_generic_acct.conf_dir modules
make[1]: Entering directory '/usr/lib/modules/5.8.12-arch1-1/build'
  CC [M]  /mnt/cold/sources/open-cas-linux/configure.d/1_generic_acct.conf_dir/test_mod.o
/mnt/cold/sources/open-cas-linux/configure.d/1_generic_acct.conf_dir/test_mod.c: In function ‘init_module’:
/mnt/cold/sources/open-cas-linux/configure.d/1_generic_acct.conf_dir/test_mod.c:7:1: error: implicit declaration of function ‘generic_start_io_acct’ [-Werror=implicit-function-declaration]
    7 | generic_start_io_acct(NULL, 0, 0, NULL);
      | ^~~~~~~~~~~~~~~~~~~~~
cc1: some warnings being treated as errors
make[2]: *** [scripts/Makefile.build:281: /mnt/cold/sources/open-cas-linux/configure.d/1_generic_acct.conf_dir/test_mod.o] Error 1
make[1]: *** [Makefile:1752: /mnt/cold/sources/open-cas-linux/configure.d/1_generic_acct.conf_dir] Error 2
make[1]: Leaving directory '/usr/lib/modules/5.8.12-arch1-1/build'
make: *** [Makefile:12: all] Error 2
make: Leaving directory '/mnt/cold/sources/open-cas-linux/configure.d/1_generic_acct.conf_dir'
### generic_start_io_acct(0, 0, NULL) ###
make: Entering directory '/mnt/cold/sources/open-cas-linux/configure.d/1_generic_acct.conf_dir'
make -C /lib/modules/5.8.12-arch1-1/build/ M=/mnt/cold/sources/open-cas-linux/configure.d/1_generic_acct.conf_dir modules
make[1]: Entering directory '/usr/lib/modules/5.8.12-arch1-1/build'
  CC [M]  /mnt/cold/sources/open-cas-linux/configure.d/1_generic_acct.conf_dir/test_mod.o
/mnt/cold/sources/open-cas-linux/configure.d/1_generic_acct.conf_dir/test_mod.c: In function ‘init_module’:
/mnt/cold/sources/open-cas-linux/configure.d/1_generic_acct.conf_dir/test_mod.c:8:1: error: implicit declaration of function ‘generic_start_io_acct’ [-Werror=implicit-function-declaration]
    8 | generic_start_io_acct(0, 0, NULL);
      | ^~~~~~~~~~~~~~~~~~~~~
cc1: some warnings being treated as errors
make[2]: *** [scripts/Makefile.build:281: /mnt/cold/sources/open-cas-linux/configure.d/1_generic_acct.conf_dir/test_mod.o] Error 1
make[1]: *** [Makefile:1752: /mnt/cold/sources/open-cas-linux/configure.d/1_generic_acct.conf_dir] Error 2
make[1]: Leaving directory '/usr/lib/modules/5.8.12-arch1-1/build'
make: *** [Makefile:12: all] Error 2
make: Leaving directory '/mnt/cold/sources/open-cas-linux/configure.d/1_generic_acct.conf_dir'
### part_round_stats(1, 1) ###
make: Entering directory '/mnt/cold/sources/open-cas-linux/configure.d/1_generic_acct.conf_dir'
make -C /lib/modules/5.8.12-arch1-1/build/ M=/mnt/cold/sources/open-cas-linux/configure.d/1_generic_acct.conf_dir modules
make[1]: Entering directory '/usr/lib/modules/5.8.12-arch1-1/build'
  CC [M]  /mnt/cold/sources/open-cas-linux/configure.d/1_generic_acct.conf_dir/test_mod.o
/mnt/cold/sources/open-cas-linux/configure.d/1_generic_acct.conf_dir/test_mod.c: In function ‘init_module’:
/mnt/cold/sources/open-cas-linux/configure.d/1_generic_acct.conf_dir/test_mod.c:9:1: error: implicit declaration of function ‘part_round_stats’ [-Werror=implicit-function-declaration]
    9 | part_round_stats(1, 1);
      | ^~~~~~~~~~~~~~~~
cc1: some warnings being treated as errors
make[2]: *** [scripts/Makefile.build:281: /mnt/cold/sources/open-cas-linux/configure.d/1_generic_acct.conf_dir/test_mod.o] Error 1
make[1]: *** [Makefile:1752: /mnt/cold/sources/open-cas-linux/configure.d/1_generic_acct.conf_dir] Error 2
make[1]: Leaving directory '/usr/lib/modules/5.8.12-arch1-1/build'
make: *** [Makefile:12: all] Error 2
make: Leaving directory '/mnt/cold/sources/open-cas-linux/configure.d/1_generic_acct.conf_dir'
### part_round_stats(NULL, 1, 1) ###
make: Entering directory '/mnt/cold/sources/open-cas-linux/configure.d/1_generic_acct.conf_dir'
make -C /lib/modules/5.8.12-arch1-1/build/ M=/mnt/cold/sources/open-cas-linux/configure.d/1_generic_acct.conf_dir modules
make[1]: Entering directory '/usr/lib/modules/5.8.12-arch1-1/build'
  CC [M]  /mnt/cold/sources/open-cas-linux/configure.d/1_generic_acct.conf_dir/test_mod.o
/mnt/cold/sources/open-cas-linux/configure.d/1_generic_acct.conf_dir/test_mod.c: In function ‘init_module’:
/mnt/cold/sources/open-cas-linux/configure.d/1_generic_acct.conf_dir/test_mod.c:10:1: error: implicit declaration of function ‘part_round_stats’ [-Werror=implicit-function-declaration]
   10 | part_round_stats(NULL, 1, 1);
      | ^~~~~~~~~~~~~~~~
cc1: some warnings being treated as errors
make[2]: *** [scripts/Makefile.build:281: /mnt/cold/sources/open-cas-linux/configure.d/1_generic_acct.conf_dir/test_mod.o] Error 1
make[1]: *** [Makefile:1752: /mnt/cold/sources/open-cas-linux/configure.d/1_generic_acct.conf_dir] Error 2
make[1]: Leaving directory '/usr/lib/modules/5.8.12-arch1-1/build'
make: *** [Makefile:12: all] Error 2
make: Leaving directory '/mnt/cold/sources/open-cas-linux/configure.d/1_generic_acct.conf_dir'
```

that is I believe the same as https://github.com/Open-CAS/open-cas-linux/issues/525

--- a/configure.d/conf_framework
+++ b/configure.d/conf_framework
@@ -7,7 +7,7 @@
 SCRIPTPATH=`dirname $0`
 SCRIPTPATH=`realpath $SCRIPTPATH`
 KERNEL_DIR="${KERNEL_DIR:-/lib/modules/$(uname -r)/build/}"
-KERNEL_VER="$(cd $KERNEL_DIR; make kernelversion)"
+KERNEL_VER="$(cd $KERNEL_DIR; make M=$SCRIPTPATH kernelversion)"
 NPROC=`nproc`
 DEFINE_FILE=$SCRIPTPATH/modules/generated_defines.h
 

